### PR TITLE
update to atmos v0.20, land v0.7, etc

### LINKED ISSUE #593: dependency updates
update to:
- [x] ClimaAtmos v0.20
- [x] RRTMGP (should remove this - we just use CA.RRTMGPI)
- [x] ClimaLSM v0.7
- [x] SurfaceFluxes v0.9
- [x] Insolation v0.9
- [x] ClimaParameters v0.8
- [x] NCDatasets v0.14

### Known interface changes
- ClimaLSM
  - coupled types/functions moved to ClimaLSM (can remove from coupler now) https://github.com/CliMA/ClimaLand.jl/pull/453
- CliMAParameters 
  - parameter setup interface changed (e.g. see https://github.com/CliMA/ClimaLand.jl/commit/acafd0a7b659e1ce31352e1f9a9ee2a56d3bf503#diff-d774aa8abce4e920dc2f5aeb7def15053f4f5a76b633effb71e699cb3b585e8f for LSM update)
- ClimaAtmos
  - get_field(sim::ClimaAtmosSimulation, ::Val{:snow_precipitation}) (see TODO in code after #567 is closed)

### CompatHelpers this will replace
- #576
- #569 
- #563 
- #559 
- #542

### PR DESCRIPTION
**Pull request by juliasloan25**
## Purpose 
update to latest dependencies 

note: ClimaLSM v0.7.0 moves almost all bucket functions dispatching on `CoupledAtmosphere` into ClimaLSM so we can remove them here. However, two functions still need to be defined in ClimaCoupler. I've documented them here: #601 

note 2: this update causes functions using the `allskywithclear` radiation option to be unstable. See https://github.com/CliMA/ClimaCoupler.jl/issues/602 for more info.

closes #593

### CODE REVIEW

File: test/flux_calculator_tests.jl [hunk 1]
```diff
@@ -44,38 +45,21 @@ function atmos_turbulent_fluxes!(sim::DummySimulation, csf)
...
-        a_h = pairs.a_h_Businger,
-        ζ_a = pairs.ζ_a_Businger,
-        γ = pairs.γ_Businger,
-    )
-    return UF.BusingerParams{FT}(; pairs...)
-end
-
-function create_parameters(toml_dict, ufpt)
+# Create parameters using CLIMAParameters.jl
+function create_parameters(toml_dict, ::UF.BusingerType)
```
Comment (nefrathenrici) [inline test/flux_calculator_tests.jl:49]: What are the param wrapper functions used for? `create_parameters`, `get_thermo_params`, etc
Each package should now own its own parameter constructor which takes in a FT or a toml dict, so you shouldn't need to make your own name map or constructor!
This should work:
```julia
import CLIMAParameters as CP
import SurfaceFluxes.Parameters as SFP
import SurfaceFluxes.UniversalFunctions as UF  # For Businger Params
import Thermodynamics.Parameters as TP

FT = Float64
sf_params = SFP.SurfaceFluxesParameters(FT, UF.BusingerParams)
thermo_params = TP.ThermodynamicsParameters(FT)

# Or with toml dict
toml_dict = CP.create_toml_dict(FT)
sf_params = SFP.SurfaceFluxesParameters(toml_dict, UF.BusingerParams)
thermo_params = TP.ThermodynamicsParameters(toml_dict)
```

File: experiments/AMIP/components/land/bucket_utils.jl [hunk 2]
```diff
@@ -50,11 +50,11 @@ get_field(sim::BucketSimulation, ::Val{:area_fraction}) = sim.area_fraction
 get_field(sim::BucketSimulation, ::Val{:air_density}) = sim.integrator.p.bucket.ρ_sfc
 
 function update_field!(sim::BucketSimulation, ::Val{:turbulent_energy_flux}, field)
-    parent(sim.integrator.p.bucket.turbulent_energy_flux) .= parent(field)
+    parent(sim.integrator.p.bucket.turbulent_fluxes.shf) .= parent(field)
```
Comment (LenkaNovak) [inline experiments/AMIP/components/land/bucket_utils.jl:53]: @juliasloan25 @kmdeck Is it better if we pass latent and sensible heat fluxes separately to ClimaLand here?

File: experiments/AMIP/components/land/bucket_utils.jl [hunk 3]
```diff
@@ -79,8 +79,8 @@ reinit!(sim::BucketSimulation) = reinit!(sim.integrator)
 # extensions required by FluxCalculator (partitioned fluxes)
 function update_turbulent_fluxes_point!(sim::BucketSimulation, fields::NamedTuple, colidx::Fields.ColumnIndex)
     (; F_turb_energy, F_turb_moisture) = fields
-    sim.integrator.p.bucket.turbulent_energy_flux[colidx] .= F_turb_energy
-    sim.integrator.p.bucket.evaporation[colidx] .=
+    sim.integrator.p.bucket.turbulent_fluxes.shf[colidx] .= F_turb_energy
+    sim.integrator.p.bucket.turbulent_fluxes.vapor_flux[colidx] .=
```
Comment (LenkaNovak) [inline experiments/AMIP/components/land/bucket_utils.jl:83]: same as above

**Review by LenkaNovak — COMMENTED** [review]
LGTM, thank you, @juliasloan25 ! It's great to see some of the parameter simplifications! I just had a comment on energy flux passing.

**Comment by LenkaNovak** [comment]
I'm also seeing some of the conservation checks failed... looking into it now

File: experiments/AMIP/components/land/bucket_utils.jl (hunk 2 above)
Comment (juliasloan25) [reply experiments/AMIP/components/land/bucket_utils.jl:53]: That would be better! Are we able to get them from Atmos?

File: experiments/AMIP/components/land/bucket_utils.jl [hunk 4]
```diff
@@ -50,11 +50,11 @@ get_field(sim::BucketSimulation, ::Val{:area_fraction}) = sim.area_fraction
...
 
 function update_field!(sim::BucketSimulation, ::Val{:turbulent_energy_flux}, field)
-    parent(sim.integrator.p.bucket.turbulent_energy_flux) .= parent(field)
+    parent(sim.integrator.p.bucket.turbulent_fluxes.shf) .= parent(field)
 end
 function update_field!(sim::BucketSimulation, ::Val{:turbulent_moisture_flux}, field)
     ρ_liq = (LSMP.ρ_cloud_liq(sim.model.parameters.earth_param_set))
-    parent(sim.integrator.p.bucket.evaporation) .= parent(field ./ ρ_liq) # TODO: account for sublimation
+    parent(sim.integrator.p.bucket.turbulent_fluxes.vapor_flux) .= parent(field ./ ρ_liq) # TODO: account for sublimation
 end
```
Comment (LenkaNovak) [inline experiments/AMIP/components/land/bucket_utils.jl:58]: Re conservation, does ClimaLand still use m / s for `sim.integrator.p.bucket.turbulent_fluxes.vapor_flux` or is it now kg / m2 / s ?

File: experiments/AMIP/components/atmosphere/climaatmos_init.jl [hunk 5]
```diff
@@ -100,11 +100,7 @@ function get_field(sim::ClimaAtmosSimulation, ::Val{:liquid_precipitation})
     sim.integrator.p.precipitation.col_integrated_rain .* ρ_liq # kg/m^2/s
```
Comment (LenkaNovak) [inline experiments/AMIP/components/atmosphere/climaatmos_init.jl:100]: Ahh, this is probably it:

```suggestion
    sim.integrator.p.precipitation.col_integrated_rain # kg/m^2/s
```

File: experiments/AMIP/components/land/bucket_utils.jl (hunk 2 above)
Comment (juliasloan25) [reply experiments/AMIP/components/land/bucket_utils.jl:53]: issue in Atmos: https://github.com/CliMA/ClimaAtmos.jl/issues/2630

File: experiments/AMIP/components/land/bucket_utils.jl (hunk 4 above)
Comment (kmdeck) [reply experiments/AMIP/components/land/bucket_utils.jl:58]: it is still m/s...that is the last thing we need to do (across all land models) to be consistent with the flux conventions we agreed on with MIT.

**Review by LenkaNovak — APPROVED** [review]
LGTM!

**Pull request merged**